request/cancellable.go

```diff
@@ -0,0 +1,23 @@
+package request
+
+import (
+	"context"
+	"net/http"
+)
+
+type CancellableRoundTripper struct {
```

Comment:
Naming: Implementation itself does not imply anything regarding canceling, only regarding context. Could we have a better name?

Also, docstring explaining what this is this would be nice